### PR TITLE
feat: persisting auth token

### PR DESCRIPTION
**Pull request by erickzanardo**
## Description

This PR adds the ability for the client to persist the token and "restore" it upon opening.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/token_provider/test/src/token_provider_test.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        clearToken: () async {},
+        getToken: () async => 'token',
+      );
+
+      final token = await tokenProvider.current;
+
+      expect(token, equals('token'));
+    });
+
+    test('returns the memory token when one', () async {
```
Comment (marwfair) [inline packages/token_provider/test/src/token_provider_test.dart:58]: You can use a different word here, but looks like this description isn't complete.
```suggestion
    test('returns the memory token when one exists', () async {
```

File: packages/token_provider/test/src/token_provider_test.dart [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      final tokenProvider = TokenProvider(
+        storeToken: (_) async {},
+        clearToken: () async {},
+        getToken: () async => 'token',
+      );
+
+      await tokenProvider.current;
```
Comment (marwfair) [inline packages/token_provider/test/src/token_provider_test.dart:65]:
```suggestion
```

Comment (erickzanardo) [reply packages/token_provider/test/src/token_provider_test.dart:65]: This is actually needed to "put the value in memory". I am not 100% happy with that, but I couldn't think in an alternative which didn't involved exposing the memory token in some way for testing.

Comment (marwfair) [reply packages/token_provider/test/src/token_provider_test.dart:65]: Gotcha. I ran the test with that line removed and everything still passed, which made me think that this line wasn't necessary.

**Review by marwfair — APPROVED** [review]
LGTM!

**Pull request merged**